app/scripts/components/analysis/results/timeseries-data.ts

```diff
@@ -254,7 +254,7 @@ async function requestTimeseries({
 
   try {
     const layerInfoFromSTAC = await queryClient.fetchQuery(
-      [TIMESERIES_DATA_BASE_ID, 'dataset', id],
+      [TIMESERIES_DATA_BASE_ID, 'dataset', id, aoi, start, end],
```

Comment:
Even though the dates were changing on the analysis page, the query key (used for caching) was not including the data, so the app was acting like nothing changed.